### PR TITLE
Adds ExtractBytes function for results

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuck@heptio.com>

### CODE REVIEW

File: pkg/results/reader.go [hunk 1]
```diff
@@ -181,6 +196,10 @@ func ExtractConfig(path string, info os.FileInfo, conf *config.Config) error {
 
 // Functions for helping with backwards compatibility
 
+func (a *Reader) MetadataDir() string {
```
Comment (timothysc) [inline pkg/results/reader.go]: Nix the "Dir" bit.

File: pkg/results/reader.go [hunk 2]
```diff
@@ -147,6 +147,21 @@ func (r *Reader) WalkFiles(walkfn filepath.WalkFunc) error {
 
 // Functions to be used within a walkfn
 
+// ExtractBytes pulls out bytes into a buffer for any path matching file.
+func ExtractBytes(file string, path string, info os.FileInfo, buf *bytes.Buffer) error {
```
Comment (davecheney) [inline pkg/results/reader.go]: try this

if file != path {
  return nil
}
reader, ok := info.Sys().(io.Reader)
 if !ok {
 		return errors.New("info.Sys() is not a reader")
 }
 _, err := buf.ReadFrom(reader)
return err

File: pkg/results/reader.go [hunk 3]
```diff
@@ -194,6 +209,10 @@ func ExtractConfig(path string, info os.FileInfo, conf *config.Config) error {
 
 // Functions for helping with backwards compatibility
 
+func (a *Reader) Metadata() string {
```
Comment (davecheney) [inline pkg/results/reader.go]: Please add a comment on this public function

Comment (timothysc) [reply pkg/results/reader.go]: Hmm that should have been caught by the build jiggery... 
/cc @liztio

Comment (liztio) [reply pkg/results/reader.go]: I think that's part of the linter, which we decided was too flaky to actually use to gate builds?

**Pull request merged**